ui/raidboss/data/06-ew/dungeon/alzadaals_legacy.ts

```diff
@@ -17,12 +46,124 @@ const triggerSet: TriggerSet<Data> = {
       response: Responses.aoe(),
     },
     {
-      id: 'Alzadaal Tentacle Dig',
-      type: 'StartsUsing',
-      netRegex: { id: ['6F55', '6559'], source: 'Ambujam', capture: false },
-      infoText: (_data, _matches, output) => output.text!(),
+      id: 'Alzadaal Tentacle Dig Collect',
+      type: 'MapEffect',
+      netRegex: { flags: tentacleFlags },
+      run: (data, matches) => data.tentacleEffects.push(matches),
+    },
+    {
+      id: 'Alzadaal Tentacle Dig Single',
+      type: 'Ability',
+      netRegex: { id: '6F55', source: 'Ambujam', capture: false },
+      delaySeconds: 2.3,
+      infoText: (data, _matches, output) => {
+        // only one tentacle (scarlet) for the first use of Tentacle Dig
+        if (data.tentacleEffects.length !== 1)
+          return output.default!();
+
+        const safeMap: { [key in Intercard]: Intercard } = {
+          NW: 'SE',
+          NE: 'SW',
+          SE: 'NW',
+          SW: 'NE',
+        };
+
+        const tentacleFlag = data.tentacleEffects[0]!.flags;
+        if (tentacleFlag === undefined || scarletTentacleLocations[tentacleFlag] === undefined)
+          return output.default!();
+        const tentacleLocation = scarletTentacleLocations[tentacleFlag]!;
+        const safeDir = safeMap[tentacleLocation];
+        if (safeDir === undefined)
+          return output.default!();
+        return output.safe!({ dir1: output[safeDir]!() });
+      },
+      run: (data) => data.tentacleEffects = [],
       outputStrings: {
-        text: {
+        NW: Outputs.dirNW,
+        NE: Outputs.dirNE,
+        SE: Outputs.dirSE,
+        SW: Outputs.dirSW,
+        safe: {
+          en: 'Go ${dir1}',
+        },
+        default: {
+          en: 'Avoid tentacle explosions',
+          de: 'Weiche Tentakel-Explosion aus',
+          fr: 'Évitez les explostions des tentacules',
+          ja: '触手の爆発から離れる',
+          cn: '躲避触手爆炸',
+          ko: '문어다리가 멈춘 곳 멀리 피하기',
+        },
+      },
+    },
+    {
+      id: 'Alzadaal Tentacle Dig Multiple',
+      type: 'Ability',
+      netRegex: { id: '6F59', source: 'Ambujam', capture: false },
+      delaySeconds: 2.3,
+      infoText: (data, _matches, output) => {
+        // both tentacles for 2nd and all subsequent uses of Tentacle Dig
+        if (data.tentacleEffects.length !== 2)
+          return output.default!();
+
+        // cyan tentacle ends on cardinal, and scarlet tentacle ends on intercard opposite cyan
+        // safe wedge runs from the cardinal adjacent to through the unoccupied intercard opposite of the cyan tentacle
+        const safeMap: {
+          [key in Cardinal]: { [idx: string]: [safecard: Cardinal, safeintercard: Intercard] };
+        } = {
+          // Cyan location: {possible Scarlet locations: [safe dirs]}
+          N: {
+            SE: ['W', 'SW'],
```

Comment:
Can these be "WSW" etc (we've used that elsewhere when we need to be more specific)?  I assume that's what you mean? Reference: <https://en.wikipedia.org/wiki/Cardinal_direction>

Reply:
Not exactly.  The problem is that the safe area is unusually shaped, which creates a safe area at a cardinal (just slightly offset) that spans through the intercardinal.  Here's an image showing the safe area for North-Northeast for example: [https://i.imgur.com/5VyHHQy.png](https://i.imgur.com/5VyHHQy.png)  (boss faces North for this mech).

NNE (or any other secondary intercardinal) is technically safe in this diagram, but so is N (partially) and NE (but far away from the boss).  So my thought was to use the double direction nomenclature (e.g. North / Northeast) to communicate both the safe cardinal (N) and the direction in which the safe zone expanded from that cardinal (NE).  Not ideal, but I think it's more accurate than just calling a secondary intercardinal (like NNE).  That said, totally open to suggestions for improving this.

Reply:
Oh! Ok that makes a lot of sense.  Thanks for the explanation.